fe/fe-core/src/main/java/com/starrocks/connector/iceberg/IcebergRewriteData.java

```diff
@@ -0,0 +1,101 @@
+// Copyright 2021-present StarRocks, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.starrocks.connector.iceberg;
+
+import com.starrocks.connector.RemoteFileInfo;
+import org.apache.iceberg.FileScanTask;
+import org.apache.iceberg.PartitionSpec;
+import org.apache.iceberg.util.StructLikeWrapper;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.NoSuchElementException;
+import java.util.stream.Collectors;
+
+
+public class IcebergRewriteData {
+
+    private IcebergConnectorScanRangeSource source;
+    private Map<StructLikeWrapper, List<TaskGroup>> partitionedTasks;
+    private int maxScanRangeLength = 100;
```

Comment:
if file scan task have more than 100, do not compaction the rest file scan task?